src/EventStore.Projections.Core/ProjectionsSubsystem.cs

```diff
@@ -213,21 +213,23 @@ public sealed class ProjectionsSubsystem :ISubsystem,
 		}
 		
 		public void Handle(ProjectionSubsystemMessage.RestartSubsystem message) {
+			var info = "PROJECTIONS SUBSYSTEM: Restarting subsystem.";
```

Comment:
Ah, sorry, one more thing. I think better not to reuse the variable 'info' in these three different case. It'll be more obvious that the right 'info' matches up to the right message if we keep the scope of the info variable smaller

Reply:
something along the lines of

```
		public void Handle(ProjectionSubsystemMessage.RestartSubsystem message) {
			if (_restarting) {
				var info = "PROJECTIONS SUBSYSTEM: Not restarting because the subsystem is already being restarted.";
				_logger.Information(info);
				message.ReplyEnvelope.ReplyWith(new ProjectionSubsystemMessage.InvalidSubsystemRestart("Restarting", info));
				return;
			}

			if (_subsystemState != SubsystemState.Started) {
				var info =
					$"PROJECTIONS SUBSYSTEM: Not restarting because the subsystem is not started. Current subsystem state: {_subsystemState}";
				_logger.Information(info);
				message.ReplyEnvelope.ReplyWith(new ProjectionSubsystemMessage.InvalidSubsystemRestart(_subsystemState.ToString(), info));
				return;
			}

			_logger.Information("PROJECTIONS SUBSYSTEM: Restarting subsystem.");
			_restarting = true;
			StopComponents();
			message.ReplyEnvelope.ReplyWith(new ProjectionSubsystemMessage.SubsystemRestarting());
		}
```

Reply:
Yes sure, let me update the scope for the info variable. Thanks a lot Tim